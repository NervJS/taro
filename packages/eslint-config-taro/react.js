module.exports = Object.assign({}, require('./index'), {
  settings: {
    react: {
      pragma: 'React',
      version: 'detect'
    }
  }
})
