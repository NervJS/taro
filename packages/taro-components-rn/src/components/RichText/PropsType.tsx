import { StyleProp, ViewStyle } from 'react-native'

export type Node = {
  type: 'node' | 'text';
  name?: string;
  attrs?: any;
  children?: Node[];
  text: string;
}

export interface RichTextProps {
  style?: StyleProp<ViewStyle>;
  nodes: Node[] | string;
}
