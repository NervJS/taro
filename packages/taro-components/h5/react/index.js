import reactifyWc from './utils/reactify-wc'
import ReactInput from './components/input'

export const View = reactifyWc('taro-view-core')
export const Icon = reactifyWc('taro-icon-core')
export const Progress = reactifyWc('taro-progress-core')
export const RichText = reactifyWc('taro-rich-text-core')
export const Text = reactifyWc('taro-text-core')
export const Button = reactifyWc('taro-button-core')
export const Checkbox = reactifyWc('taro-checkbox-core')
export const CheckboxGroup = reactifyWc('taro-checkbox-group-core')
export const Editor = reactifyWc('taro-editor-core')
export const Form = reactifyWc('taro-form-core')
export const Input = ReactInput
export const Label = reactifyWc('taro-label-core')
export const Picker = reactifyWc('taro-picker-core')
export const PickerView = reactifyWc('taro-picker-view-core')
export const PickerViewColumn = reactifyWc('taro-picker-view-column-core')
export const Radio = reactifyWc('taro-radio-core')
export const RadioGroup = reactifyWc('taro-radio-group-core')
export const Slider = reactifyWc('taro-slider-core')
export const Switch = reactifyWc('taro-switch-core')
export const CoverImage = reactifyWc('taro-cover-image-core')
export const Textarea = reactifyWc('taro-textarea-core')
export const CoverView = reactifyWc('taro-cover-view-core')
export const MovableArea = reactifyWc('taro-movable-area-core')
export const MovableView = reactifyWc('taro-movable-view-core')
export const ScrollView = reactifyWc('taro-scroll-view-core')
export const Swiper = reactifyWc('taro-swiper-core')
export const SwiperItem = reactifyWc('taro-swiper-item-core')
export const FunctionalPageNavigator = reactifyWc('taro-functional-page-navigator-core')
export const Navigator = reactifyWc('taro-navigator-core')
export const Audio = reactifyWc('taro-audio-core')
export const Camera = reactifyWc('taro-camera-core')
export const Image = reactifyWc('taro-image-core')
export const LivePlayer = reactifyWc('taro-live-player-core')
export const Video = reactifyWc('taro-video-core')
export const Map = reactifyWc('taro-map-core')
export const Canvas = reactifyWc('taro-canvas-core')
export const Ad = reactifyWc('taro-ad-core')
export const OfficialAccount = reactifyWc('taro-official-account-core')
export const OpenData = reactifyWc('taro-open-data-core')
export const WebView = reactifyWc('taro-web-view-core')
export const NavigationBar = reactifyWc('taro-navigation-bar-core')
export const Block = reactifyWc('taro-block-core')
export const CustomWrapper = reactifyWc('taro-custom-wrapper-core')
