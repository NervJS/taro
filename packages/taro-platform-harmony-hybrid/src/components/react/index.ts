/* eslint-disable simple-import-sort/exports */
export { Ad } from '@tarojs/components/lib/react'
export { AdCustom } from '@tarojs/components/lib/react'
export { AnimationVideo } from '@tarojs/components/lib/react'
export { AnimationView } from '@tarojs/components/lib/react'
export { ArCamera } from '@tarojs/components/lib/react'
export { Audio } from '@tarojs/components/lib/react'
export { AwemeData } from '@tarojs/components/lib/react'
export { Block } from '@tarojs/components/lib/react'
export { Button } from '@tarojs/components/lib/react'
export { Camera } from '@tarojs/components/lib/react'
export { Canvas } from '@tarojs/components/lib/react'
export { ChannelLive } from '@tarojs/components/lib/react'
export { ChannelVideo } from '@tarojs/components/lib/react'
export { Checkbox, CheckboxGroup } from '@tarojs/components/lib/react'
export { CommentDetail, CommentList } from '@tarojs/components/lib/react'
export { ContactButton } from '@tarojs/components/lib/react'
export { CoverImage } from '@tarojs/components/lib/react'
export { CoverView } from '@tarojs/components/lib/react'
export { CustomWrapper } from '@tarojs/components/lib/react'
export { Editor } from '@tarojs/components/lib/react'
export { FollowSwan } from '@tarojs/components/lib/react'
export { Form } from '@tarojs/components/lib/react'
export { FunctionalPageNavigator } from '@tarojs/components/lib/react'
export { Icon } from '@tarojs/components/lib/react'
export { Image } from '@tarojs/components/lib/react'
export { InlinePaymentPanel } from '@tarojs/components/lib/react'
export { Input } from '@tarojs/components/lib/react'
export { KeyboardAccessory } from '@tarojs/components/lib/react'
export { Label } from '@tarojs/components/lib/react'
// export { Label } from './components/label/label'
export { Lifestyle } from '@tarojs/components/lib/react'
export { Like } from '@tarojs/components/lib/react'
export { LivePlayer } from '@tarojs/components/lib/react'
export { LivePusher } from '@tarojs/components/lib/react'
export { Login } from '@tarojs/components/lib/react'
export { Lottie } from '@tarojs/components/lib/react'
export { Map } from '@tarojs/components/lib/react'
export { MovableArea, MovableView } from '@tarojs/components/lib/react'
export { NavigationBar } from '@tarojs/components/lib/react'
export { Navigator } from '@tarojs/components/lib/react'
export { OfficialAccount } from '@tarojs/components/lib/react'
export { OpenData } from '@tarojs/components/lib/react'
export { PageMeta } from '@tarojs/components/lib/react'
export { Picker, PickerGroup } from '@tarojs/components/lib/react'
export { PickerView, PickerViewColumn } from '@tarojs/components/lib/react'
export { Progress } from '@tarojs/components/lib/react'
export { PullToRefresh } from '@tarojs/components/lib/react'
export { Radio, RadioGroup } from '@tarojs/components/lib/react'
export { RichText } from '@tarojs/components/lib/react'
export { RtcRoom, RtcRoomItem } from '@tarojs/components/lib/react'
export { ScrollView } from '@tarojs/components/lib/react'
export { ShareElement } from '@tarojs/components/lib/react'
export { Slider } from '@tarojs/components/lib/react'
export { NativeSlot, Slot } from '@tarojs/components/lib/react'
export { Swiper, SwiperItem } from '@tarojs/components/lib/react'
export { Switch } from '@tarojs/components/lib/react'
export { Tabbar, TabItem } from '@tarojs/components/lib/react'
export { Tabs } from '@tarojs/components/lib/react'
export { Text } from '@tarojs/components/lib/react'
export { Textarea } from '@tarojs/components/lib/react'
export { Video, VideoControl, VideoDanmu } from '@tarojs/components/lib/react'
export { View } from '@tarojs/components/lib/react'
export { VoipRoom } from '@tarojs/components/lib/react'
export { WebView } from '@tarojs/components/lib/react'
// export * from './components/block/block'
// export * from './components/grid/grid'
// export * from './components/root-portal/root-portal'
// export * from './components/sticky-header/sticky-header'
// export * from './components/sticky-section/sticky-section'
// export * from './components/editor/editor'
// export { default as ListView } from './components/list/list'
// export { MatchMedia } from './components/match-media/match-media'
// export { default as PageContainer } from './components/page-container/page-container'
export { StickyHeader, StickySection, RootPortal, GridView, ListView, MatchMedia, PageContainer } from '@tarojs/components/lib/react'
