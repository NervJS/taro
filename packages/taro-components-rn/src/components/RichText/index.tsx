/**
 * âœ” nodes
 */

import * as React from 'react'
import {
  View,
  WebView,
} from 'react-native'
import * as ReactDOMServer from 'react-dom/server.browser'
import { omit, parseStyles } from '../../utils'
import { RichTextProps, Node } from './PropsType'

class _RichText extends React.Component<RichTextProps> {
  static defaultProps = {
    nodes: ''
  }

  renderChildrens = (arr: Array<any> = []) => {
    if (arr.length === 0) return
    return arr.map((list) => {
      if (list.type === 'text') {
        return this.renderText(list.text)
      }
      return this.renderNodes(list)
    })
  }

  renderText = (text: string = '') => {
    return React.createElement('span', {
      dangerouslySetInnerHTML: { __html: text },
      key: Math.random()
    })
  }

  renderNodes = (item: Node): React.ReactElement => {
    const child = this.renderChildrens(item.children)
    return React.createElement(
      item.name || 'div',
      {
        key: Math.random(),
        ...omit(item.attrs, ['class', 'style']),
        className: item.attrs.class,
        style: parseStyles(item.attrs.style)
      },
      child
    )
  }

  render () {
    const {
      style,
      nodes,
    } = this.props

    const html: string = typeof nodes === 'string'
      ? nodes
      : nodes.map((item: Node): string => {
        return ReactDOMServer.renderToStaticMarkup(this.renderNodes(item))
      }).join(',')

    return (
      <View style={style}>
        <WebView
          source={{ html }}
          scalesPageToFit={false}
          injectedJavaScript={`
            document.documentElement.style.padding = 0;
            document.documentElement.style.margin = 0;
            document.body.style.padding = 0;
            document.body.style.margin = 0;
            true;
          `}
          style={{
            backgroundColor: 'transparent'
          }}
        />
      </View>
    )
  }
}

export default _RichText
