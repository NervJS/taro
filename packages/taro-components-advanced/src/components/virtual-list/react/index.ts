import { ScrollView, View } from '@tarojs/components'
import React from 'react'

import List from './list'
import outerWrapper from './wrapper'

import type { VirtualListProps } from '../'
import type { IProps } from '../preset'

const VirtualList = React.forwardRef(function VirtualList (props: VirtualListProps, ref) {
  const {
    direction = 'ltr',
    outerElementType = ScrollView,
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanCount = 1,
    ...rest
  } = props as IProps

  if ('children' in rest) {
    console.warn('Taro(VirtualList): children props have been deprecated. ' + 'Please use the item props instead.')
    rest.item = rest.children as IProps['item']
  }
  if (rest.item instanceof Array) {
    console.warn('Taro(VirtualList): item should not be an array')
    rest.item = rest.item[0]
  }
  return React.createElement(List, {
    ref,
    ...rest,
    outerElementType,
    itemElementType,
    innerElementType,
    direction,
    initialScrollOffset,
    overscanCount,
    outerWrapper,
  })
})

export default VirtualList
