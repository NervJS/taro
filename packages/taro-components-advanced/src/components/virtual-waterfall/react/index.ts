import { ScrollView, View } from '@tarojs/components'
import React from 'react'

import Waterfall from './waterfall'
import outerWrapper from './wrapper'

import type { VirtualWaterfallProps } from '../'
import type { IProps } from '../preset'

const VirtualWaterfall = React.forwardRef(function VirtualWaterfall (props: VirtualWaterfallProps, ref) {
  const {
    outerElementType = ScrollView,
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanDistance = 50,
    ...rest
  } = props as IProps

  if ('children' in rest) {
    console.warn('Taro(VirtualWaterfall): children props have been deprecated. ' + 'Please use the item props instead.')
    rest.item = rest.children as IProps['item']
  }
  if (rest.item instanceof Array) {
    console.warn('Taro(VirtualWaterfall): item should not be an array')
    rest.item = rest.item[0]
  }
  return React.createElement(Waterfall, {
    ref,
    ...rest,
    outerElementType,
    itemElementType,
    innerElementType,
    initialScrollOffset,
    overscanDistance,
    outerWrapper,
  })
})

export default VirtualWaterfall
