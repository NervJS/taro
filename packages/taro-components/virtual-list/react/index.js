import React from 'react'
import FixedSizeList from './FixedSizeList'
import { ScrollView, View } from '@tarojs/components'

function convertPxToInt (style) {
  if (typeof style === 'string') {
    const str = style.toLowerCase()
    if (/px$/.test(str)) {
      return Number(str.replace(/px$/, ''))
    }
  }
  return style
}

const OuterScrollView = React.forwardRef(
  (props, ref) => {
    const { style, onScroll, onScrollNative, layout, ...rest } = props
    const handleScroll = event => {
      onScroll({
        ...event,
        currentTarget: {
          ...event.detail,
          clientWidth: convertPxToInt(style.width),
          clientHeight: convertPxToInt(style.height)
        }
      })

      if (typeof onScrollNative === 'function') {
        onScrollNative(event)
      }
    }

    return React.createElement(ScrollView, {
      ref,
      style,
      scrollY: layout === 'vertical',
      scrollX: layout === 'horizontal',
      onScroll: handleScroll,
      ...rest
    })
  }
)

const VirtualList = React.forwardRef((props, ref) => {
  const {
    direction = 'ltr',
    innerElementType = View,
    itemElementType = View,
    initialScrollOffset = 0,
    overscanCount = 1,
    ...rest
  } = props

  return React.createElement(FixedSizeList, {
    ref,
    ...rest,
    itemElementType,
    innerElementType,
    outerElementType: OuterScrollView,
    direction,
    initialScrollOffset,
    overscanCount
  })
})

export default VirtualList
