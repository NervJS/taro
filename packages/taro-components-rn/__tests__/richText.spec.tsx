// eslint-disable-next-line no-use-before-define
import * as React from 'react'
import renderer from 'react-test-renderer'
import RichText from '../src/components/RichText'

describe('RichText', () => {
  it('RichText render', () => {
    const nodes: any = [
      {
        name: 'div',
        type: 'node',
        attrs: {
          class: 'div_class',
          style: `
            line-height: 60px;
            color: white;
            font-size: 60px;
          `
        },
        children: [
          {
            type: 'text',
            text: 'Hello World!'
          }
        ]
      }
    ]
    const tree = renderer.create(<RichText nodes={nodes} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
