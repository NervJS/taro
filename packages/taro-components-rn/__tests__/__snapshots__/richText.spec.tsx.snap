// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RichText RichText render 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <RNCWebView
      cacheEnabled={true}
      injectedJavaScript="
            document.documentElement.style.padding = 0;
            document.documentElement.style.margin = 0;
            document.body.style.padding = 0;
            document.body.style.margin = 0;
            true;
          "
      injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
      injectedJavaScriptForMainFrameOnly={true}
      javaScriptEnabled={true}
      messagingEnabled={false}
      onContentProcessDidTerminate={[Function]}
      onHttpError={[Function]}
      onLoadingError={[Function]}
      onLoadingFinish={[Function]}
      onLoadingProgress={[Function]}
      onLoadingStart={[Function]}
      onMessage={[Function]}
      onShouldStartLoadWithRequest={[Function]}
      scalesPageToFit={false}
      source={
        Object {
          "html": "<div class=\\"div_class\\" style=\\"line-height:60px;color:white;font-size:60px\\"><span>Hello World!</span></div>",
        }
      }
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#ffffff",
          },
          Object {
            "backgroundColor": "transparent",
          },
        ]
      }
      useSharedProcessPool={true}
    />
  </View>
</View>
`;
